Libraries/PeriphDrivers/Source/SPIMSS/spimss_reva.c

```diff
@@ -231,6 +241,203 @@ int MXC_SPIMSS_RevA_MasterTrans(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *
+    int spi_num = MXC_SPIMSS_GET_IDX((mxc_spimss_regs_t *)spi);
+    int ret_val = E_NO_ERROR;
+
+    //Reset static variables.
+    memset(&dma_config_tx, 0, sizeof(dma_config_tx));
+    memset(&dma_config_rx, 0, sizeof(dma_config_rx));
+    memset(&srcdst_config_tx, 0, sizeof(srcdst_config_tx));
+    memset(&srcdst_config_rx, 0, sizeof(srcdst_config_rx));
+    memset(&dma_adv_config_tx, 0, sizeof(dma_adv_config_tx));
+    memset(&dma_adv_config_rx, 0, sizeof(dma_adv_config_rx));
+
+    //We use SPIMSS master trans function to send data from
+    ret_val = MXC_SPIMSS_RevA_TransSetup(spi, req, 1);
+    if (E_NO_ERROR != ret_val) {
+        return ret_val;
+    }
+
+    //Setting dmas fifo levels for tx and rx in spimss.
+    spi->dma &= ~(MXC_F_SPIMSS_DMA_RX_FIFO_LVL);
+    spi->dma &= ~(MXC_F_SPIMSS_DMA_TX_FIFO_LVL);
+
+    //Initialization of DMA
+    if (true == states[spi_num].auto_dma_handlers) {
+        MXC_DMA_Init();
+    }
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelTx)) {
+        //Getting the available tx channel id.
+        tx_channel_id = MXC_DMA_AcquireChannel();
+        if (tx_channel_id < 0) {
+            return tx_channel_id;
+        }
+        MXC_SPIMSS_RevA_SetTXDMAChannel(spi, tx_channel_id);
+
+    } else {
+        tx_channel_id = MXC_SPIMSS_RevA_GetTXDMAChannel(spi);
+        if (tx_channel_id < 0) {
+            return E_BAD_STATE;
+        }
+    }
+    //Enable NVIC interrupt of the dma channel which will be used as TX channel.
+    MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(tx_channel_id), MXC_DMA_Handler);
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelRx)) {
+        rx_channel_id = MXC_DMA_AcquireChannel();
+        if (rx_channel_id < 0) {
+            return rx_channel_id;
+        }
+        MXC_SPIMSS_RevA_SetRXDMAChannel(spi, rx_channel_id);
+    } else {
+        rx_channel_id = MXC_SPIMSS_RevA_GetRXDMAChannel(spi);
+        if (rx_channel_id < 0) {
+            return E_BAD_STATE;
+        }
+    }
+    //Enable NVIC interrupt of the dma channel which will be used as RX channel.
+    MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(rx_channel_id), MXC_DMA_Handler);
+
+    states[spi_num].req = req;
+
+    // TX dma configuration settings.
+    dma_config_tx.ch = tx_channel_id;
+    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPI1TX;
+    dma_config_tx.dstwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_tx.srcwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_tx.dstinc_en = 0; // Ignored since the destination is set to SPIMSS TX fifo.
+    dma_config_tx.srcinc_en = 1;
+
+    // advanced config for tx channel
+    dma_adv_config_tx.ch = tx_channel_id;
+    dma_adv_config_tx.burst_size = req->bits < 16 ? 1 : 2;
+
+    // advanced config for rx channel
+    dma_adv_config_rx.ch = rx_channel_id;
+    dma_adv_config_rx.burst_size = req->bits < 16 ? 1 : 2;
+
+    dma_config_rx.ch = rx_channel_id;
+    dma_config_rx.reqsel = MXC_DMA_REQUEST_SPI1RX;
```

Comment:
Same DMA request select issue as mentioned above.

Reply:
I simply added these macro controls:

For tx:
#if TARGET_NUM == 32660
dma_config_tx.reqsel = MXC_DMA_REQUEST_SPI1TX;
#elif TARGET_NUM == 32650
dma_config_tx.reqsel = MXC_DMA_REQUEST_SPIMSSTX;
#else
return E_NOT_SUPPORTED;
#endif

For rx:
#if TARGET_NUM == 32660
dma_config_rx.reqsel = MXC_DMA_REQUEST_SPI1RX;
#elif TARGET_NUM == 32650
dma_config_tx.reqsel = MXC_DMA_REQUEST_SPIMSSRX;
#else
return E_NOT_SUPPORTED;
#endif